source/app-development/tutorials-interactive-apps/k8s-like-hpc-jupyter.rst

```diff
@@ -0,0 +1,371 @@
+.. _app-development-tutorials-interactive-apps-k8s-like-hpc-jupyter:
+
+Add a Jupyter App on a Kubernetes Cluster that behaves like HPC compute
+=======================================================================
+
+This tutorial will walk you through creating an interactive Jupyter app that
+your users will use to launch a `Jupyter Notebook Server`_ in a kubernetes cluster.
+The container will behave much like a HPC compute node. This has the benefit that
+a single app can serve both traditional HPC as well as Kubernetes.
+
+It assumes you have a working understanding of app development already. The purpose of
+this document is to describe how to write apps specifically for a kubernetes cluster,
+so it skips a lot of important details about app development that may be found in
+other tutorials like :ref:`app-development-tutorials-interactive-apps-add-jupyter`.
+
+
+We're going to be looking at the `bc osc jupyter`_ app which you can fork, clone
+and modify for your site.  This page also holds the `submit yml in full`_ for reference.
+
+The container
+-------------
+
+The container to make Kubernetes pods look like HPC compute would end up needing all
+the OS packages present on the HPC environment.  The OS of the container itself would also
+need to compatible with packages installed on HPC environment so if you run RHEL 8 on HPC
+you would also need to run RHEL 8 inside the container.
+
+Things like Lmod and HPC applications will need to be run inside the Pod's container just like
+if a job was spawned in the traditional HPC resource manager.
+
+
+Switch between SLURM and Kubernetes
+-----------------------------------
+
+The first big change from a traditional HPC interactive app is the main YAML structure is wrapped
+in a large ``if`` statement based on the cluster choice. If a user choose one of the HPC clusters,
+the SLURM submit YAML is rendered, otherwise the Kubernetes YAML is rendered.
+
+Here is the beginning of the block:
+
+.. code-block:: text
+
+  <% if cluster =~ /owens|pitzer/ -%>
+  ---
+  batch_connect:
+    template: "basic"
+    conn_params:
+      - jupyter_api
+  script:
+  ...SLURM specific...
+
+
+Here is the logic to select Kubernetes:
+
+.. code-block:: text
+
+  <% elsif cluster =~ /kubernetes/
+  ...Ruby variables setup here...
+
+container spec
+--------------
+
+Let's look at this section first of the Kubernetes block.  Here you must specify the ``name``, ``image``
+and ``command``.  The name determines the Pod Id (the job name in HPC parlance).
+
+The ``image`` should be the HPC container image and ``command`` will be the job script that has been adapted to
+work with both SLURM and Kubernetes.  The ``command`` will be run from the user's home directory and will cover mount
+requirements in :ref:`mount requirements <kubernetes-mount-requirements>`.
```

Comment:
Don't know how to word this off hand, but: 

The `command` would be the same for every site & app and is the exact same script that would be ran if the job were ran on Slurm. Which means it runs `before.sh`, `script.sh`, `after.sh` and any other template files just like apps ran on other HPC clusters.

This point I think will help them conceptualize what's happening here. Same script, same entrypoint just totally different environment (kinda/not really after all the setup).

Reply:
Added a paragraph about how the script is generated using same files as traditional HPC